api.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"net/http"
+
+	"github.com/cyverse-de/logcabin"
+)
+
+// API represents a single instance of the REST API.
+type API struct {
+	emailClient *EmailClient
+	deSettings  *DESettings
+}
+
+func NewAPI(emailClient *EmailClient, deSettings *DESettings) *API {
+	return &API{
+		emailClient: emailClient,
+		deSettings:  deSettings,
+	}
+}
+
+// handles email notification requests
+func (a *API) EmailRequestHandler(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/" {
+		http.Error(w, "404 not found.", http.StatusNotFound)
+		return
+	}
+	switch r.Method {
+	case "GET":
+		w.WriteHeader(200)
+		w.Write([]byte("A service that handles email-requests and send out emails to users."))
+	case "POST":
+		logcabin.Info.Println("Post request received.")
+		emailReq, payloadMap, err := parseRequestBody(r)
+		if err != nil {
+			logcabin.Error.Println(err)
+			//w.WriteHeader(getErrorResponseCode(err))
+			//w.Write([]byte(err.Error()))
+			JSONError(w, r, (string(err.Error())), getErrorResponseCode(err))
```

Comment:
I mentioned this in the other PR as well, but I thought I'd mention it again just because the code was moved to a new file. The type cast can be removed if you like.